zap/src/main/java/org/zaproxy/zap/view/ContextIncludePanel.java

```diff
@@ -64,6 +75,56 @@ public static String getPanelName(Context context) {
         return getPanelName(context.getId());
     }
 
+    private File selectImportFile() {
+        JFileChooser chooser = new JFileChooser(Constant.getContextsDir());
+        File file = null;
+
+        int rc = chooser.showOpenDialog(View.getSingleton().getMainFrame());
+        if (rc == JFileChooser.APPROVE_OPTION) {
+            file = chooser.getSelectedFile();
+            if (file == null || !file.exists()) {
+                return null;
+            }
+            return file;
+        }
+        return null;
+    }
+
+    private void addImportButton() {
+        String IMPORT_BUTTON_LABEL = Constant.messages.getString("context.include.import.label");
+        JButton importButton = new JButton(IMPORT_BUTTON_LABEL);
+        importButton.addActionListener(
+                new ActionListener() {
+                    @Override
+                    public void actionPerformed(ActionEvent evt) {
+                        File file = selectImportFile();
+                        if (file != null) {
+                            List<String> regexes = regexesPanel.getRegexes();
+
+                            String line = null;
+                            try (BufferedReader br = new BufferedReader(new FileReader(file))) {
+                                while ((line = br.readLine()) != null) {
+                                    String trimmedRegex = line.trim();
+                                    if (!regexes.contains(trimmedRegex)) {
+                                        Pattern.compile(trimmedRegex, Pattern.CASE_INSENSITIVE);
+                                        regexes.add(trimmedRegex);
+                                    }
+                                }
+                            } catch (IOException | PatternSyntaxException e) {
+                                View.getSingleton()
+                                        .showWarningDialog(
+                                                Constant.messages.getString(
+                                                        "context.include.import.error",
+                                                        e.getLocalizedMessage()));
+                                return;
+                            }
+                            regexesPanel.setRegexes(regexes);
```

Comment:
Note: I haven't tested.

Won't this replace what is already specified? If the user has already set something or imported another file first.

Reply:
Existing regexes are loaded in :101 and new ones from file are added (excluding duplicates).